### DISCUSSION TITLE
How to deal with `createdAt` and `updatedAt` field in a prisma model?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [prisma/prisma#20323] [Q&A] [answered]
I use sequelize generated the model:
```cmd
npx sequelize model:generate --name DispsysProj --attributes proj_name:string,proj_dir:string,proj_version:string,proj_desc:text  
```

In the schema.prisma, it is this:
```prisma
model DispsysProjs {
  id           Int      @id @default(autoincrement())
  proj_name    String?  @db.VarChar(255)
  proj_dir     String?  @db.VarChar(255)
  proj_version String?  @db.VarChar(255)
  proj_desc    String?
  createdAt    DateTime @db.Timestamptz(6)
  updatedAt    DateTime @db.Timestamptz(6)
}
```
I have a question about it,
when I insert one record of the table:
there says `createdAt` and `updatedAt` must have data:
```
Argument createdAt for data.0.createdAt is missing.
Argument updatedAt for data.0.updatedAt is missing.
```

to my way of thinking, the `createdAt` and `updatedAt` should be add by prisma, or they could be optional, so how can I do with it?
could I change to :
```prisma
  createdAt?    DateTime @db.Timestamptz(6)
  updatedAt?    DateTime @db.Timestamptz(6)
```

**Comment by nurul3101** [comment, marked as answer]
Hello @liaodalin19903 👋 

You would need to use the `@default` and` @updatedAt` attributes in your Prisma model.

The `@default` attribute sets a default value for a field each time a new record is created. You can use it with `now()` to set the current timestamp as the default value. ([Reference](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#default))

The `@updatedAt` attribute updates the timestamp each time the record is updated. ([Reference](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#updatedat))

Your model would look something like this:

```prisma
model DispsysProjs {
  id           Int      @id @default(autoincrement())
  proj_name    String?  @db.VarChar(255)
  proj_dir     String?  @db.VarChar(255)
  proj_version String?  @db.VarChar(255)
  proj_desc    String?
  createdAt    DateTime @default(now()) @db.Timestamptz(6)
  updatedAt    DateTime @updatedAt @db.Timestamptz(6)
}
```

With these changes, you should not need to provide values for createdAt and updatedAt when you create or update records in the DispsysProjs table. Prisma will manage these fields for you.

Please let us know in case you have any other queries.
It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved.

**Reply by nurul3101** [reply]
@liaodalin19903 Checking to see if you had a chance to look at my last response?

Please let us know in case you still need assistance.

**Reply by ckblockit** [reply]
Hi, I decided to  manually add this after instrospect our existing database.
However, whenever I try to create a new migration off of this baseline, it complains the following message.

[*] Changed the `contacts` table
  [*] Altered column `createdAt` (default changed from `Some(Now)` to `None`)
  
<img width="735" alt="image" src="https://github.com/user-attachments/assets/1ea55585-2a36-4686-9de4-5c67960e3487" />
